### DISCUSSION TITLE
Co-contributors Discussion

### DISCUSSION THREAD

**Discussion by natintosh** [natintosh/intl-phone-number-input#201] [Ideas]
Hello, recently, I have been having troubles maintaining and publishing new updates for this library. The library has grown larger than I expected and now and then, new ideas are popping up here and there. It now becoming more and more difficult to support this library because of my inexperience and tight schedules. 

I am willing to open up this library to **co-contributors** who have amazing ideas to add to this library and support this library. 

If you are interested in becoming a **co-contributor**, please feel free to mention it in the discussion below, thank you.

**Comment by ShivamJaglan** [comment]
I am using this library and would like to be a co-contributor.

**Reply by natintosh** [reply]
Nice!

**Comment by cedvdb** [comment]
Hello @natintosh I've been working on a dart version of lib phone number. It is not a 1 to 1 conversion. This can be useful for you because it will instanty give you support for web and android. 

https://pub.dev/packages/phone_numbers_parser (a like woud)

In my honnest opinion you are doing too much in this library by doing the libphonenumber channeling thing + the translation. This lib might be more maintainable by being UI only, and the translation shouldn't be in a single json (I think it is, pardon me if it isn't) as it makes the build size bigger. The translation could be left to the user to do given a template file.

I hope my library can help you, I put a lot of work into it so a like will be appreciated.

**Reply by natintosh** [reply]
Awesome 👍 I'll check this out. I'm kind of in-between my exams, so once I'm done I'll look into it.

**Comment by omtodkar** [comment]
hey @natintosh,

I recently integrated this library, I just wanted to say, before you are looking for contributors you should actively review, comment if necessary and merge existing pull requests from all people, eventually you can also get few maintaners from those PR authors as they have already worked on this library.

No offence, but while I was looking for a fix for an issue (which in one of the pending PRs), I came across this pinned post so shared my thoughts.

**Reply by natintosh** [reply]
None taken. It's really hard maintaining projects, especially open-source projects. I truly try to merge PRs and fix issues whenever I am able to but when you feel you are finally done, another PR or issue pops up 🥶.

This month, I've been busy with school works mostly exams so I went offline to focus on my studies which ended yesterday with the intention of coming back to the backlogs of tasks waiting to be fixed or merge. I did receive lots of notification during that period but there is little I could do. 

Notwithstanding, this is the reason why I would be happy to work with other talents as co-contributors/co-maintainers. I actually don't mind giving other talents access to this repository and also pass to publish this package on pub.dev.

**Comment by MasudCodes** [comment]
Hi , is there any way to change the text " Search by Country Code "  ? , because I recently got one of my app rejected from Huawei App Store because in that list of country code ,  Hong Kong was available . So I have to change the text to Country/Region . 

Thank you

**Comment by FarhanSyedain** [comment]
Hey there! I want to work on this library. There are still many things which user does not have control on like the textstyle of searchbox and otherthings.

**Comment by ctykaya** [comment]
I like this library and want to be contributor also.

**Reply by natintosh** [reply]
Awesome. I'd like to onboard you

**Reply by FarhanSyedain** [reply]
I too wanna contribute.

**Comment by Bringoff** [comment]
Well, we use the library in our company and have already opened one PR. I've planned to create another one. But being co-contributor might simplify the things.

**Comment by dreking** [comment]
I use this library a lot. Would love to contribute as well

**Comment by gauravgp25** [comment]
I am using this and it is so well and would like to be a co-contributor.

**Comment by 1cornerstone** [comment]
I used this package recently but the decoration I need isn't there so I fork and made a pull request. Also, I will like to join the co-contributor

**Reply by bynalab** [reply]
Thanks @1cornerstone Have you made PR on your changes?

**Reply by 1cornerstone** [reply]
Yes . I did

**Reply by bynalab** [reply]
Please point me to it.

**Comment by 1cornerstone** [comment]
https://github.com/natintosh/intl_phone_number_input/pull/357

**Comment by deandreamatias** [comment]
Hi!
I'm currently the main contributor of [form_builder_phone_field](https://github.com/flutter-form-builder-ecosystem/form_builder_phone_field) and [phone_number](https://github.com/flutter-form-builder-ecosystem/phone_number).
I think that would be nice join efforts to made a nice package/plugin to validate phone numbers, on the air or by form field

To my have been so hard maintain [phone_number](https://github.com/flutter-form-builder-ecosystem/phone_number), for several reasons.
Perhaps we can consider deprecating some package. I will write my thoughts about the all packages:

1. [form_builder_phone_field](https://github.com/flutter-form-builder-ecosystem/form_builder_phone_field): Depends on [phone_number](https://github.com/flutter-form-builder-ecosystem/phone_number). Works ok, but don't have support to web.
2. [phone_number](https://github.com/flutter-form-builder-ecosystem/phone_number): the original owner was @nashfive and transfer the property to [FlutterFormBuilder Ecosystem](https://github.com/flutter-form-builder-ecosystem) on last year. I made some updates but is hard to maintain because the dependencies and that I don't have expertise on there code and plugins in general.
3. [intl_phone_number_input](https://github.com/natintosh/intl_phone_number_input): This is a popular package/plugin. You have two parts here:
-  Phone validation and format (utils): Is a nice implemention with support to web, using libphonenumber_plugin. I think that is important maintain this part
- Field phone input (widgets): Widgets is really hard to maintain because has a lot of properties and customization than developers request to have. In the same way, this part is no ready-to-use with packages forms, like reactive_forms or flutter_form_builder. I don't think that is profit on maintain this part. We can offer possibilities if you decide depreciated this part.
4. [phone_numbers_parser](https://pub.dev/packages/phone_numbers_parser) from @cedvdb: Is a really nice package and a alternative to packages that use libphonenumber like plugin. I think that phone_numbers_parser can coexist with other packages that use libphonenumber_plugin, and offer alternatives to developers

This is my thoughts only, isn't a mandatory thing on any way, of course. We can think alternatives together if you want.

**Comment by gauravgp25** [comment]
Hi,
I want to join and put effort to make a package and to make better
alternatives.

Please accept my request and take me with you for development.
